vispy/visuals/surface_plot.py

```diff
@@ -107,16 +107,24 @@ def set_data(self, x=None, y=None, z=None, colors=None):
                     x = np.arange(self._z.shape[0])
                 else:
                     x = self._x
-            self.__vertices[:, :, 0] = x.reshape(len(x), 1)
+            if x.ndim == 1:
+                self.__vertices[:, :, 0] = x.reshape(len(x), 1)
+
+            if x.ndim == 2:
+                self.__vertices[:, :, 0] = x  # Copy the 2D data into the appropriate slice
             update_mesh = True
 
-        if new_vertices or y is not None:
-            if y is None:
-                if self._y is None:
-                    y = np.arange(self._z.shape[1])
-                else:
-                    y = self._y
-            self.__vertices[:, :, 1] = y.reshape(1, len(y))
+            if new_vertices or y is not None:
+                if y is None:
+                    if self._y is None:
+                        y = np.arange(self._z.shape[1])
+                    else:
+                        y = self._y
+    
+                if y.ndim == 1:
+                    self.__vertices[:, :, 1] = y.reshape(1, len(y))
+                elif y.ndim == 2:
+                    self.__vertices[:, :, 1] = y  # Copy the 2D data into the appropriate slice
```

Comment:
This could probably be simplified to:

```suggestion
                if y.ndim == 1:
                    y = y.reshape(1, len(y))
                self.__vertices[:, :, 1] = y  # Copy the 2D data into the appropriate slice
```

Does that still work? Right?

Reply:
Same could be done for the `x` portion of the code above too.

Reply:
It may not work

Reply:
It works!!! Both for x and y

Reply:
Great! If you want to include it in this PR then all you have to do is push a new commit to this same branch (your master branch). In the future it would be a good idea to create a separate branch for your pull requests, but since you've already created this let's just go with it.

Reply:
Do you want me to change it to the way you are proposing? Leaving it the way I made would improve readability

Reply:
> Leaving it the way I made would improve readability

Depends who's reading the code. Your version will only support 1D and 2D arrays and will ignore anything else. This will be confusing to users who pass 3D (or higher) arrays as they probably won't receive an error message and their visual will still get drawn. Mine changes the code to represent what it is doing: converting `y` from a 1D to a 2D shape and only assigns to the output once. Mine will also produce an error message if `y` can't be broadcast into the vertices array.

Reply:
Done